test/conformance/tests/sriov_operator.go

```diff
@@ -995,50 +995,90 @@ var _ = Describe("[sriov] operator", func() {
 			Context("Resource Injector", func() {
 				// 25815
 				It("Should inject downward api volume", func() {
-					node := sriovInfos.Nodes[0]
-					intf, err := sriovInfos.FindOneSriovDevice(node)
-					Expect(err).ToNot(HaveOccurred())
+					resourceName := "apivolresource"
+					sriovNetworkName := "test-apivolnetwork"
+					numVfs := 5
+					var node string
+					var intf *sriovv1.InterfaceExt
 
-					nodePolicy := &sriovv1.SriovNetworkNodePolicy{
-						ObjectMeta: metav1.ObjectMeta{
-							GenerateName: "test-apivolumepolicy",
-							Namespace:    operatorNamespace,
-						},
+					if discovery.Enabled() {
+						var numVfs int
+						var err error
+						node, resourceName, numVfs, intf, err = discovery.DiscoveredResources(clients,
+							sriovInfos, operatorNamespace,
 
-						Spec: sriovv1.SriovNetworkNodePolicySpec{
-							NodeSelector: map[string]string{
-								"kubernetes.io/hostname": node,
+							func(policy sriovv1.SriovNetworkNodePolicy) bool {
+								if !defaultFilterPolicy(policy) {
+									return false
+								}
+								if &policy.Spec.NicSelector == nil || &policy.Spec.NicSelector.PfNames == nil || len(policy.Spec.NicSelector.PfNames) != 1 {
```

Comment:
Why do we need the check of the nicselector (and why we did not need it before in the generic policy case) ?
We should accept policies with other types of selector (I think pfNames is the less likely to be used)

Reply:
This was a prerequisite for finding a specific nic. Removed, as this is not really needed in this test.

Reply:
I think this whole test can go under the generic config category, I am not seeing any custom configuration here.